src/DataStructures/Matrix.hpp

```diff
@@ -66,3 +77,17 @@ bool operator==(const Matrix& lhs, const Matrix& rhs);
 
 /// Inequivalence operator for Matrix
 bool operator!=(const Matrix& lhs, const Matrix& rhs);
+
+// ---
+// Algebra
+
+/// Matrix multiplication
+Matrix operator,(const Matrix& lhs, const Matrix& rhs);
```

Comment:
Never overload the comma operator.

I'm not quite sure what we want to do here.  Using * for matrix-matrix multiplication would work fine, but we don't want to do that for matrix-vector because it would interact poorly with the DataVector*DataVector operation.  We have a `dot_product` function for Tensors, so maybe we could use overloads of that.

Reply:
I like the idea of overloading dot_product